src/common.rs

```diff
@@ -24,3 +24,36 @@ fn remove_invalid_leading_chars_test() {
     let string = String::from("***FOO-BAR");
     assert_eq!("FOO-BAR", remove_invalid_leading_chars(&string));
 }
+
+pub(crate) mod tests {
+    use super::*;
+    use std::path::PathBuf;
+
+    #[allow(dead_code)]
+    pub fn blank_line_entry(number: usize, total_lines: usize) -> LineEntry {
+        LineEntry {
+            number,
+            file: FileEntry {
+                path: PathBuf::from(".env"),
+                file_name: ".env".to_string(),
+                total_lines,
+            },
+            raw_string: String::from("\n"),
+            is_deleted: false,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn line_entry(number: usize, total_lines: usize, str: &str) -> LineEntry {
```

Comment:
I have a couple questions/suggestions for my own knowledge here:

1. Is there a reason to use a ``line_entry()``` function instead of a ```LineEntry::new()``` constructor?
2. I don't think it's bad to use ```str``` as an argument (it's not a reserved keyword or anything to my knowledge, but maybe confusing since it's also kind of related to a type?), but maybe the final argument could be called ```raw_string``` since that is the actual struct property's name?

Reply:
> I have a couple questions/suggestions for my own knowledge here:
> 
> 1. Is there a reason to use a `line_entry()` function instead of a `LineEntry::new()` constructor? I can think of a reason to use a constructor (keep associated functionality of struct in the same place), but I'm wondering if there's a reason to leave it as a standalone function?

We don't need to have the constructor function `LineEntry::new()`, because we create `LineEntry` only in two places.
The `line_entry()` function is needed only for tests and it localed inside the `tests` module.

> 2. I don't think it's bad to use `str` as an argument (it's not a reserved keyword or anything to my knowledge, but maybe confusing since it's also kind of related to a type?), but maybe the final argument could be called `raw_string` since that is the actual struct property's name?

I agree 👍